console-csharp-snippets-sample/AuthenticationHelper.cs

```diff
@@ -133,25 +124,21 @@ public static GraphServiceClient GetAuthenticatedClientForApp()
         public static async Task<string> GetTokenForAppAsync()
         {
             AuthenticationResult authResult;
-
             authResult = await IdentityAppOnlyApp.AcquireTokenForClientAsync(new string[] { "https://graph.microsoft.com/.default" });
-            TokenForApp = authResult.AccessToken;
-
-            return TokenForApp;
+            return authResult.AccessToken;
         }
 
         /// <summary>
         /// Signs the user out of the service.
         /// </summary>
-        public static void SignOut()
+        public static async void SignOut()
         {
-            foreach (var user in IdentityClientApp.Users)
+            IEnumerable<IAccount> accounts = await IdentityClientApp.GetAccountsAsync();
+            foreach (var account in accounts.ToArray())
```

Comment:
Use the ToArray() to avoid a CollectionChangedException.

Reply:
Ok, I've made this change.